### PR TITLE
action: implements base of circulation.

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds checkin and checkout action.
* Adds renew and hold action.

Co-Authored-by: Lauren-D <laurent.dubois@uclouvain.be>

### CODE REVIEW

**Review by benerken — APPROVED** [review]

File: invenio_sip2/actions/actions.py [hunk 1]
```diff
@@ -265,6 +266,7 @@ class BlockPatron(Action):
     @check_selfcheck_authentication
     def execute(self, message, client, **kwargs):
         """Execute action."""
+        # TODO: implements action
         return
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:270]: Cosmetic : `pass` ?

File: invenio_sip2/actions/actions.py [hunk 2]
```diff
@@ -273,16 +275,75 @@ class Checkin(Action):
...
+        patron_id = message.get_field_value('patron_id')
+
+        checkin = checkin_handler(
+            client.remote_app, client.user_id, client.institution_id,
+            patron_id, item_id, language=patron_session.get('language')
+        )
+
+        # prepare message based on required fields
+        response_message = self.prepare_message_response(
+            ok='1' if checkin.is_success else '0',
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:290]: `ok = str(int(checkin.is_success))`

File: invenio_sip2/actions/actions.py [hunk 3]
```diff
@@ -273,16 +275,75 @@ class Checkin(Action):
...
+        patron_id = message.get_field_value('patron_id')
+
+        checkout = checkout_handler(
+            client.remote_app, client.user_id, client.institution_id,
+            patron_id, item_id, language=patron_session.get('language')
+        )
+
+        # prepare message based on required fields
+        response_message = self.prepare_message_response(
+            ok='1' if checkout.is_success else '0',
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:327]: same here

File: invenio_sip2/actions/actions.py [hunk 4]
```diff
@@ -291,6 +352,7 @@ class FeePaid(Action):
     @check_selfcheck_authentication
     def execute(self, message, client, **kwargs):
         """Execute action."""
+        # TODO: implements action
         return
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:356]: Cosmetic : `pass` ?

File: invenio_sip2/actions/actions.py [hunk 5]
```diff
@@ -299,17 +361,73 @@ class Hold(Action):
...
+        patron_id = message.get_field_value('patron_id')
+
+        hold = hold_handler(
+            client.remote_app, client.user_id, client.institution_id,
+            patron_id, item_id, language=patron_session.get('language')
+        )
+
+        # prepare message based on required fields
+        response_message = self.prepare_message_response(
+            ok='1' if hold.is_success else '0',
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:376]: same

File: invenio_sip2/actions/actions.py [hunk 6]
```diff
@@ -299,17 +361,73 @@ class Hold(Action):
...
+        patron_id = message.get_field_value('patron_id')
+
+        renew = renew_handler(
+            client.remote_app, client.user_id, client.institution_id,
+            patron_id, item_id, language=patron_session.get('language')
+        )
+
+        # prepare message based on required fields
+        response_message = self.prepare_message_response(
+            ok='1' if renew.is_success else '0',
```
Comment (zannkukai) [inline invenio_sip2/actions/actions.py:411]: :-)

**Review by zannkukai — APPROVED** [review]
Code will works. Remarks are for a better "pure" code

File: invenio_sip2/actions/actions.py (hunk 5 above)
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: ok

File: invenio_sip2/actions/actions.py (hunk 6 above)
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: ok

File: invenio_sip2/actions/actions.py (hunk 3 above)
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: ok

File: invenio_sip2/actions/actions.py (hunk 2 above)
Comment (lauren-d) [reply invenio_sip2/actions/actions.py]: thanks. it's done

**Pull request merged**